shm/inc/coordinates.hpp

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include <cstddef>

```

Comment:
Why do you need <cstddef> in the header? You generally want to have as little includes in header in favour of cpp includes. This way you avoid importing libraries to file you do not use with header.

Reply:
The header is needed for definition of 'size_t' type (according to description in https://en.cppreference.com/w/cpp/types/size_t). Somehow without it the code wouldn't compile (in my gcc compiler).

Reply:
Sorry. My error